pkg/agent/controller/l7flowexporter/l7_flow_export_controller.go

```diff
@@ -87,7 +87,7 @@ func NewL7FlowExporterController(
 		namespaceInformer:     namespaceInformer.Informer(),
 		namespaceLister:       namespaceInformer.Lister(),
 		namespaceListerSynced: namespaceInformer.Informer().HasSynced,
-		l7Reconciler:          l7Reconciler,
+		startSuricataOnceFn:   l7Reconciler.StartSuricataOnce,
```

Comment:
should we just replace the `l7Reconciler` parameter with a `func() error` parameter?

Reply:
`l7Reconciler` is used only for starting Suricata with method `StartSuricataOnce`. I remove `l7Reconciler` here and add a global variable `startSuricataOnceFn` which can be overridden for testing.